lib/pbench/server/auth/__init__.py

```diff
@@ -115,18 +115,24 @@ def set_well_known_endpoints(self):
             )
             raise
 
-    def get_oidc_public_key(self, token: str):
+    def set_oidc_public_key(self):
+        """Sets the public key of the current OIDC client"""
+
+        realm_public_key_uri = f"{self.server_url}/realms/{self.realm_name}"
+        public_key = (
+            "-----BEGIN PUBLIC KEY-----\n"
+            + self._get(realm_public_key_uri).json()["public_key"]
+            + "\n-----END PUBLIC KEY-----"
+        )
```

Comment:
I share a similar concern.  Perhaps something like:

```
try:
    response = self._get(realm_public_key_uri)
except ... < exception handling as appropriate > ...:
    ...
try:
    public_key = response.json()["public_key"]
except KeyError:
    < appropriate error handling>
OpenIDCClient.PUBLIC_KEY = f"""-----BEGIN PUBLIC KEY-----
{public_key}
-----END PUBLIC KEY-----
"""

Reply:
The exception from `_get` won't fly free, the exceptions will be propagated from the `_get` method and It is up to the caller of the `set_oidc_public_key` to implement the appropriate try-except block for it (mostly that would be inside the shell). I plan to do this as part of OIDC client initialization with #3056 

I will add the doc string to suggest what kind of exceptions this function could raise.

Reply:
I'm inclined to side with @npalaska on this one.  I'm concerned that, in general, we're investing too much code in handling errors at the locations where they occur instead of deferring that to where the error matters.  I understand the motivations behind this inclination, and I have great sympathy for them, but I think we would be better off if we let the pendulum swing back a bit.

This function doesn't have any invariants which it has to maintain -- all it has to do is set a value or notify its caller that it failed -- so, I don't think it needs any error handling in the body of the function:  it will either work, or it will raise an exception for the caller to deal with.

Put another way, in @portante's sample code above, what is supposed to go in the `appropriate error handling` blocks?  If it's nothing more than a bare `raise`, then we don't need the `try` blocks to begin with.

Reply:
I wasn't objecting: I was pointing out a potential issue and making sure it had been considered. The problem is a matter of defining the failure modes and their scope and impact. Documentation can help to clarify these things.